packages/with_bloc/lib/src/with_bloc.dart

```diff
@@ -22,6 +22,13 @@ class WithBloc<BlocType extends ValueNotifier<StateType>, StateType>
   ///
   /// This function might be called multiple times in the lifetime of this widget.
   /// Whenever the [inputs] change, a new BLoC will be created by calling this function.
+  ///
+  /// For creating the initial bloc, this function will be called in the `initState`.
+  /// That means that you can listen to any `Provider`s or `MediaQuery`.
+  ///
+  /// If you need to `listen` to changes, consider moving the `Provider` call out of the method
+  /// and then using the `value` itself in the [createBloc].
+  /// Don't forget to also reference the `value` in the [inputs].
```

Comment:
Should we add a line like:

> If the value of the `Provider` is a `ValueListenable`, you don't have to listen through the provider, but can subscribe to the value itself.

The above is a little rough, but the point would be that people don't start moving stuff outside of the `WithBloc.createBloc`when it's not strictly necessary.

Reply:
I think the start of the sentence makes this clear with:
> If you need to `listen` to changes

Reply:
But what I want to suggest to the reader, is not listening on the widget-level and using `inputs` to recreate the bloc, but rather take the full `ValueNotifier<T>` in the constructor of the bloc and subscribe inside the bloc itself. (So the bloc doesn't get recreated all the time.)